backend/src/main/java/com/htecgroup/skynest/controller/CompanyController.java

```diff
@@ -0,0 +1,168 @@
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"Access denied\"],"
+                                + " \"status\": \"403\","
+                                + " \"timestamp\": \"2022-06-07 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "404",
+            description = "Tier doesn't exist",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"Tier doesn't exist\"],"
+                                + " \"status\": \"409\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "409",
+            description = "Pib, email or phone number already in use",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"This email is already in use\"],"
+                                + " \"status\": \"409\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = String.class),
+                  examples = {@ExampleObject(value = "Internal Server Error")})
+            })
+      })
+  @io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content =
+          @Content(
+              schema = @Schema(implementation = CompanyIO.class),
+              mediaType = "application/json",
+              examples = {
+                @ExampleObject(
+                    value =
+                        "{\"email\": \"username@gmail.com\","
+                            + " \"pib\": \"123456789\","
+                            + "  \"name\": \"Name\","
+                            + "  \"tierName\": \"basic\","
+                            + "  \"phoneNumber\": \"38166575757\","
+                            + "  \"address\": \"Local address\"}")
+              }))
+  @PreAuthorize("hasAuthority(T(com.htecgroup.skynest.model.entity.RoleEntity).ROLE_ADMIN)")
+  @PostMapping
+  public ResponseEntity<CompanyIO> addCompany(@Valid @RequestBody CompanyIO companyIO) {
```

Comment:
Does this validation work since you don't have Validated annotation on class level?

Reply:
Works, @Valid in the list of arguments is enough.